### PR TITLE
leverage platform extensions

### PR DESCRIPTION
**Pull request by chirag04**
leverage platform extension files so that they are not unnecessarily included in the bundle.

### CODE REVIEW

**Comment by chirag04** [comment]
cc @DanielMSchmidt @satya164

File: src/index.android.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+import { DrawerLayoutAndroid } from 'react-native';
```
Comment (satya164) [inline src/index.android.js:1]: This should work

`export { DrawerLayoutAndroid as default } from 'react-native';`

File: src/index.ios.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+import DrawerLayout from 'react-native-drawer-layout';
```
Comment (satya164) [inline src/index.ios.js]: This should work (not sure)

`export from 'react-native-drawer-layout';`

or

`export { default as default } from 'react-native-drawer-layout';`

Comment (chirag04) [reply src/index.ios.js]: `babel-plugin-transform-export-extensions` is not available by default in RN.

Comment (satya164) [reply src/index.ios.js]: `export { default as default } from 'react-native-drawer-layout';` should not require any plugin

Comment (chirag04) [reply src/index.ios.js]: I just double checked. that does not work. i would leave it as is.

**Review by DanielMSchmidt — CHANGES_REQUESTED** [review]
We had it like this before but this would be a problem for windows code. I think leaving it as it is would be the option with the least amount of duplicate code, right?

**Comment by chirag04** [comment]
I can add a .windows.js file if that's supported

**Comment by satya164** [comment]
Why not `index.js` and `index.android.js` though? Also feels weird to mix common js export and es6 import. Any reason it's not ES6 export?

**Comment by satya164** [comment]
Also feels weird to mix common js export and es6 import. Any reason it's not ES6 export?

File: src/index.ios.js (hunk 2 above)
Comment (satya164) [reply src/index.ios.js]: What happens? This is a valid export syntax in ES2015 and works as expected for me with just `es2015` preset.

**Comment by chirag04** [comment]
@satya164 I didn't realize packager prioritizes `.platform.js` over `.js`? About the import, yeah. I rushed this little bit. I can fix those.

**Comment by satya164** [comment]
Yes, the packager will look for `index.{ios|android}.js`, `index.native.js` and `index.js`.

**Comment by DanielMSchmidt** [comment]
I would be in favour of a `index.android.js` and a `index.js` file, this sounds nice. Also we should switch to ES6 exports and also change the readme to state that for people using the commonjs syntax. Thank you so much for your help @chirag04 @satya164 !

File: src/index.android.js (hunk 1 above)
Comment (DanielMSchmidt) [reply src/index.android.js:1]: Wow, this would be so neat!

**Comment by chirag04** [comment]
ok. addressed the feedback

**Review by DanielMSchmidt — APPROVED** [review]
Thank you for this PR, I will test it out later today and ship it afterwards

**Comment by chirag04** [comment]
@DanielMSchmidt let's merge after this is merged and bumped: https://github.com/react-native-community/react-native-drawer-layout/pull/17

I can bump the dependency for drawer-layout after that one is merged.

**Comment by DanielMSchmidt** [comment]
Yeah, good idea!

**Pull request merged**